### PR TITLE
Salesforce Sandbox

### PR DESCRIPTION
**Pull request by nick-Ag**
Adds a `isSandbox` global setting to Salesforce, the setting is used to toggle the login url between `login.salesforce` and `test.salesforce`.

App PR: https://github.com/segmentio/app/pull/13019
Oauth-service PR: https://github.com/segmentio/oauth-service/pull/199

## Testing

Tested successfully in stage.

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/salesforce/index.ts [hunk 1]
```diff
@@ -26,6 +26,11 @@ const destination: DestinationDefinition<Settings> = {
           'The user specific instance URL returned by Salesforce Oauth. This setting is hidden to the user and set by Oauth Service.',
         type: 'string',
         required: true
+      },
+      isSandbox: {
+        label: 'Sandbox Instance',
+        description: 'Turn this on if you would like to oauth into a sandbox environment instead',
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/salesforce/index.ts:32]: You are using `.` to end other sentences. Did you miss one here?

Comment (nick-Ag) [reply packages/destination-actions/src/destinations/salesforce/index.ts:32]: I think I'm just blind 🙈

**Review by marinhero — APPROVED** [review]

File: packages/destination-actions/src/destinations/salesforce/index.ts [hunk 2]
```diff
@@ -26,6 +26,11 @@ const destination: DestinationDefinition<Settings> = {
           'The user specific instance URL returned by Salesforce Oauth. This setting is hidden to the user and set by Oauth Service.',
         type: 'string',
         required: true
+      },
+      isSandbox: {
```
Comment (brennan) [inline packages/destination-actions/src/destinations/salesforce/index.ts:30]: Do we need to regenerate types so this setting appears in the `Settings` interface?

**Review by marinhero — CHANGES_REQUESTED** [review]
Brennan is right, we need to regenerate the types here.

File: packages/destination-actions/src/destinations/salesforce/index.ts [hunk 3]
```diff
@@ -26,6 +26,11 @@ const destination: DestinationDefinition<Settings> = {
...
+        label: 'Sandbox Instance',
+        description: 'Turn this on if you would like to oauth into a sandbox environment instead.',
```
Comment (mericsson) [inline packages/destination-actions/src/destinations/salesforce/index.ts:32]:
```suggestion
        description: 'Enable to authenticate into a sandbox instance.',
```

**Review by marinhero — APPROVED** [review]

**Pull request merged**